src/api/lock.jl

```diff
@@ -0,0 +1,81 @@
+const liblock = ReentrantLock()
+
+@static if VERSION >= v"1.6"
+    """
+        HDF5.API.use_api_lock::Bool
+
+    Julia compile time preference of whether to use a `ReentrantLock`. By default
+    the HDF5 C library is not thread-safe. Concurrent calls to the HDF5 library
+    from multiple threads will likely fail.
+    """
+    const use_api_lock = Preferences.@load_preference("use_api_lock", default = true)
```

Comment:
```suggestion
    const use_api_lock::Bool = Preferences.@load_preference("use_api_lock", default = true)
```
then the error below is not necessary

Reply:
My thinking here is that the error I wrote would be more descriptive than the following since this potentially faces an end user. I suppose I could try catch the MethodError though.
```julia
ERROR: MethodError: Cannot `convert` an object of type String to an object of type Bool
Closest candidates are:
  convert(::Type{T}, ::Ptr) where T<:Integer at pointer.jl:23
  convert(::Type{T}, ::T) where T<:Number at number.jl:6
  convert(::Type{T}, ::Number) where T<:Number at number.jl:7
```

Reply:
I guess? Seems like a bit of a weird error case to be worried about though.

Reply:
Actually the type assertion a global will not work with Julia 1.6, and I think we're still trying to maintain compat with Julia 1.3